### PR TITLE
Draft-commit of materials data structure

### PR DESCRIPTION
**Pull request by Marvellanius**
So this weekend I didn't find the time to work on this, but found some time today.

This is clearly a draft, with incomplete data to facilitate concrete discussion about the preferred structure for the materials data.

The proposed way should still conform to the currently defined routes, so no changes to those are necessary, which I figured was preferable.

Yeah not much else to say, I think the changes speak for themselves, but please name anything that could be improved regarding structure, naming of keys, etc.

By the way, I chose to use incomplete data to convey the idea of the structure, naturally when a final structure is decided, I'll fill in the missing data

### CODE REVIEW

**Comment by JMPJNS** [comment]
should we add which characters/weapons use the materials?

**Comment by NurMarvin** [comment]
> 
> 
> should we add which characters/weapons use the materials?
that would be pretty nice, yeah, though I'm unsure how much work that would be

**Comment by Marvellanius** [comment]
Possibly, could add related characters and weapons to relevant items (e.g. `characters: [], weapons: []`).
what I've been wondering is how we'd do the linking. We could either drop in the full name (which would be less preferred imho for accessing it programmatically), or drop in the 'slug' -as I like to call it- in this case, the foldername (e.g. 'albedo' instead of 'Albedo').

Now to make consuming the API easier for rendering purposes we could also choose to go with an array of objects: `characters:[{name: "Albedo", slug: "albedo"}, {name: "Diona", slug: "diona"}]` but to be fair, I would only do this to reduce API requests, and since all of the responses get cached anyway, that might not be a concern.
So my preferred way of 'linking' would probably go to using the slug for characters and weapons

File: assets/data/materials/boss-material/en.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "anemo": {
+        "slug": "hurricane-seed",
```
Comment (Marvellanius) [inline assets/data/materials/boss-material/en.json:3]: Slug might not be the correct key for this, as we're pretty much using this as the unique identifier for all other entities. Would `id` be a more proper way to name this field (I would then change this across all files in this pull request)

Comment (NurMarvin) [reply assets/data/materials/boss-material/en.json:3]: Yeah, I think using `id` would be better in this case.

**Review by NurMarvin — APPROVED** [review]

**Pull request merged**